frontend/src/components/GameMap/Activity.js

```diff
@@ -1,56 +1,64 @@
 import React, { useEffect, useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
 import { getActivityImg } from '../../utils/constants';
 import { Square, ActivityImg } from './ActivityStyles';
+import { PageRoutes } from '../../utils/constants';
 
 function Activity(props) {
     const gameSquare = useRef();
+    const navigate = useNavigate();
 
     useEffect(() => {
         const updateSize = () => {
             // * 0.90 - prevents rows wrapping
-            gameSquare.current.style.width = props.parent.current.offsetWidth / props.mapSizeY * 0.90 + "px";
-            gameSquare.current.style.height = props.parent.current.offsetHeight / props.mapSizeX * 0.90 + "px";
-        }
+            if (gameSquare !== null) {
```

Comment:
```suggestion
             if (gameSquare != null && gameSquare.current != null) {
```
Aktualnie sypie się tu i trzeba zamienić na takie warunki w if'ie